test/functional/mempool_package_limits.py

```diff
@@ -26,7 +26,7 @@ def run_test(self):
         self.wallet = MiniWallet(self.nodes[0])
         # Add enough mature utxos to the wallet so that all txs spend confirmed coins.
         self.generate(self.wallet, 35)
-        self.generate(self.nodes[0], COINBASE_MATURITY)
+        self.generate(self.wallet, COINBASE_MATURITY)
```

Comment:
The MiniWallet does not update the confirmation counts of its UTXOs when new blocks are generated. When the blocks are generated with a different generator, it will still think that its generated coins are immature even if enough blocks have been mined to make them mature. When this behavior is coupled with the change to prefer matured coins, the MiniWallet ends up later having a very restricted UTXO set as it will have a few mature coins that are the result of making transaction chains. This occurs because of the fallback to use immature coins when no mature coins are available.

By generating the maturity blocks with the MiniWallet, we end up having it rescan for its UTXOs which makes it update the confirmation counts of its coins thereby actually making the mature coins set as full as expected. It is possible to achieve the same result by explicitly calling `self.wallet.rescan_utxos()` after generating the blocks.

Incidentally, this indicates that there might be an intermittent failure in this test as it may randomly select some UTXOs for package building that were spent by chain building loops. This is readily apparent when the mature UTXO pool is small.